src/System.Runtime.InteropServices/tests/System/Runtime/InteropServices/Marshal/StructureToPtrTests.cs

```diff
@@ -0,0 +1,186 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Xunit;
+
+namespace System.Runtime.InteropServices.Tests
+{
+    public class StructureToPtrTests
+    {
+        public struct StructWithIntField
+        {
+            public int value;
+        }
+
+        public struct StructWithByValArray
+        {
+            [MarshalAs(UnmanagedType.ByValArray, SizeConst = 5)]
+            public StructWithIntField[] array;
+        }
+
+        public struct StructWithBoolArray
+        {
+            [MarshalAs(UnmanagedType.ByValArray, SizeConst = 2)]
+            public bool[] array;
+        }
+
+        public struct StructWithDateArray
+        {
+            [MarshalAs(UnmanagedType.ByValArray, SizeConst = 2)]
+            public DateTime[] array;
+        }
+
+        [StructLayout(LayoutKind.Auto)]
+        public struct SomeTestStruct_Auto
+        {
+            public int i;
+        }
+
+        [Fact]
+        public void AutoLayoutStructureTest()
+        {
+            var someTs_Auto = new SomeTestStruct_Auto();
+            ArgumentException ex = AssertExtensions.Throws<ArgumentException>("structure", () => Marshal.StructureToPtr(someTs_Auto, new IntPtr(123), true));
+            Assert.Contains("The specified structure must be blittable or have layout information.", ex.Message);
```

Comment:
This still seems brittle validation if we were to localize these messages.  We've generally stayed away from such message validation as a result.

Reply:
I've deleted the Assert about the message string.